src/content/en/2020/third-parties.md

```diff
@@ -8,16 +8,261 @@ authors: [simonhearne]
+  caption="Pages with advertising third parties are more likely to have a high CPU load time.",
+  description="Distribution of CPU load time by presence of third party categories. Most categories follow the same pattern, with advertising the outlier showing higher CPU load time, especially at lower percentiles.",
+  width=600,
+  height=371
+  )
+}}
+
+This chart shows the probability density function of total page CPU time by the third-party categories present on each page. The median page is at 50 on the percentile axis. The data shows that all third-party categories follow a similar pattern, with the median page between 400 - 1,000ms CPU time. The outlier here is advertising (in black): if a page has advertising tags it is much more likely to have high CPU usage during page load. The median page with advertising tags has a CPU load time of 1,500ms, compared to 500ms for pages without advertising. The high CPU load time at the lower percentiles indicates that even the fastest sites are impacted significantly by the presence of third-parties categorized as advertising.
+
+## Other
+### Timing-Allow-Origin prevalence
+
+The [Resource Timing API] allows website owners to measure the performance of individual resources via JavaScript. This data is, by default, extremely limited for cross-origin resources like third-party content. There are legitimate reasons for not providing this timing information such as responses that vary by authentication state: e.g. a website owner may be able to determine if a visitor is logged into a Facebook by measuring the response size of a widget request. For most third-party content, though, setting the timing-allow-origin header is an act of transparency to allow the hosting website to track performance and size of their third-party content.
+
+{{ figure_markup(
+  image="requests-with-tao.png",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSNXusoFJKi8Z1i_yuG_5umUCJOZwBMaZaECq4T8lU5zIQuLR7UHfZOJmXsXzPWQTJFnFa3dcOEPJgy/pubchart?oid=1886505312&format=interactive",
+  caption="Less than 35% of third party responses are served with a timing-allow-origin header",
+  description="Less than 35% of third party responses are served with a timing-allow-origin header",
+  width=600,
+  height=371
+  )
+}}
+
+The results in HTTP Archive show that only one third of third-party responses expose detailed size and timing information to the hosting website.
+
+## Repercussions
+
+We know that adding arbitrary JavaScript to our sites introduces risks to both site speed and security. Site owners must be diligent to balance the value of the third-party scripts they include with the speed penalty they may bring, and use modern features such as [subresource integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) and [content security policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) to maintain a strong security posture.
```

Comment:
Might be worth referencing the security chapter here, since that should cover both SRI and CSP adoption.